etc/kayobe/ansible/deploy-radosgw-usage-exporter.yml

```diff
@@ -0,0 +1,152 @@
+          OS_CACERT: "{{ '/etc/ssl/certs/ca-certificates.crt' if os_distribution == 'ubuntu' else '/etc/pki/tls/certs/ca-bundle.crt' }}"
+      when: stackhpc_enable_radosgw_usage_exporter
+
+    - name: Check ec2 credential for admin
+      command: >
+        {{ venv }}/bin/openstack
+        ec2 credentials list --user admin
+        --format json
+      environment: "{{ openstack_auth_env }}"
+      register: credential_check
+      delegate_to: "{{ groups['controllers'][0] }}"
+      vars:
+        ansible_host: "{{ hostvars[groups['controllers'][0]].ansible_host }}"
+      run_once: true
+      when: stackhpc_enable_radosgw_usage_exporter
+
+    - name: Create ec2 credential if there's none
+      command: >
+        {{ venv }}/bin/openstack
+        ec2 credentials create --user admin
+        --format json
+      environment: "{{ openstack_auth_env }}"
+      delegate_to: "{{ groups['controllers'][0] }}"
+      vars:
+        ansible_host: "{{ hostvars[groups['controllers'][0]].ansible_host }}"
+      run_once: true
+      when:
+        - stackhpc_enable_radosgw_usage_exporter
+        - "{{ credential_check.stdout == [] }}"
+
+    - name: Query ec2 credential for admin
+      command: >
+        {{ venv }}/bin/openstack
+        ec2 credentials list --user admin
+        --format json
+      environment: "{{ openstack_auth_env }}"
+      delegate_to: "{{ groups['controllers'][0] }}"
+      vars:
+        ansible_host: "{{ hostvars[groups['controllers'][0]].ansible_host }}"
+      register: credential
+      run_once: true
+      when: stackhpc_enable_radosgw_usage_exporter
+
+    - name: Get object storage endpoint
+      command: >
+        {{ venv }}/bin/openstack
+        endpoint list --service object-store --interface internal
+        --format json
+      environment: "{{ openstack_auth_env }}"
+      delegate_to: "{{ groups['controllers'][0] }}"
+      vars:
+        ansible_host: "{{ hostvars[groups['controllers'][0]].ansible_host }}"
+      register: endpoint
+      run_once: true
+      when: stackhpc_enable_radosgw_usage_exporter
+
+    - name: Ensure radosgw_usage_exporter container is running
+      community.docker.docker_container:
+        name: radosgw_usage_exporter
+        image: ghcr.io/stackhpc/radosgw_usage_exporter:0.1.0
+        network_mode: host
+        env:
+          RADOSGW_SERVER: "{{ radosgw_server }}"
+          ADMIN_ENTRY: admin
+          ACCESS_KEY: "{{ ec2.Access }}"
+          SECRET_KEY: "{{ ec2.Secret }}"
+          VIRTUAL_PORT: "{{ stackhpc_radosgw_usage_exporter_port | string }}"
+        entrypoint: "{{ [ 'python', '-u', './radosgw_usage_exporter.py', '--insecure' ] if not stackhpc_radosgw_usage_exporter_verify else omit }}"
+      vars:
+        ec2: "{{ credential.stdout | from_json | first }}"
+        host: "{{ endpoint.stdout | from_json | first }}"
+        radosgw_server: "{{ host.URL | regex_replace('(https?://)([^:/]+):([0-9]+)/.*', '\\1\\2:\\3') }}" # Drop Swift api address
```

Comment:
Generally I'm not a fan of gnarly regex like this but I understand in some places it's necessary. Can you think of any alternatives here?

Reply:
I found we can use urlsplit ansible filter!